lib/tests/test_scratch_buffers.c

```diff
@@ -153,7 +155,26 @@ Test(scratch_buffers, stats_counters_are_updated)
       /* check if byte counter is updated */
       scratch_buffers_update_stats();
       cr_assert_eq(stats_counter_get(stats_scratch_buffers_bytes), count * DEFAULT_ALLOC_SIZE);
+
+      /* Check internal state is as expected */
+      cr_assert(scratch_buffers_get_local_allocation_count() == count,
+                "Local allocation count is not updated properly, value=%d, expected=%d",
+                scratch_buffers_get_local_allocation_count(), count);
     }
+
+  scratch_buffers_explicit_gc();
+  cr_assert(scratch_buffers_get_local_usage_count() == 0,
+            "get_local_usage_count() failed to reset to 0, value=%d, expected=%d",
+            scratch_buffers_get_local_usage_count(), 0);
+  cr_assert(scratch_buffers_get_local_allocation_count() == count,
+            "Local allocation count failed to reset to 0, value=%d, expected=%d",
+            scratch_buffers_get_local_allocation_count(), count);
+
+  scratch_buffers_allocator_deinit();
+  count = 0;
+  cr_assert(stats_counter_get(stats_scratch_buffers_count) == count,
```

Comment:
You could just simply assert against `== 0` and not reset `count` to 0. While both are correct, the later (setting count to 0 and assert if equals to count) makes harder to understand what is expected here.

Upon reading the test one could remember that count represents the number of time the `scratch_buffers_alloc` is called, and checking out this assert make not much sense. Of course it can be seen from close by code, but imho when asserting on equality the expected value should be simple, by simple I mean if possible not computed by the test.

Reply:
done.